python/sitecustomize.py

```diff
@@ -0,0 +1,4 @@
+import coverage
+
+# This call will initialize coverage for each Python process (but only if COVERAGE_PROCESS_START is set).
+coverage.process_startup()
```

Comment:
Is there any other way to do this? Which processes exactly need to run this code? Would there be some other way to hook them? I'm worried about including `sitecustomize` here because:

1. It may not load in all testing configurations because the source directly isn't always in the path.
2. It may shadow or be shadowed by a system level `sitecustomize` module. And this shadowing would happen even when coverage is not in use.

Reply:
Interestingly this is the recommend way.  An alternative is to create a .pth file in the Python installation.  We can go the alternative way if there is a good reason.

Note that this invocation does nothing if the env variable is not set:
```
    cps = os.environ.get("COVERAGE_PROCESS_START")
    if not cps:
        # No request for coverage, nothing to do.                                                                                                                                              
        return None
```

(response to 1) the setup is for running python tests (with server in background). This should be the default configuration for which we collect Python coverage.  (I feel that is the only testing configuration of interest at the moment, and we can revise things later if more configurations end up being needed.)

(response to 2) I do no worry if this file is shadowed as we would figure that out when collecting coverage.  However, if this file can prevent execution of something else that could be a problem (but can it?)

Reply:
I'm pretty sure it can prevent execution of a locally installed `sitecustomize.py`. However, maybe we don't care. We can continue with this setup if needed.